src/pages/data/container/SortLayout.tsx

```diff
@@ -25,57 +29,33 @@ export default function SortLayout() {
         className={defaultFilterStyle}
         optionValue={C.majorOptions}
         onValueChange={(value) => {
-          setSelectedMajor(value);
-          setSelectedMiddle('');
-          setSelectedMinor('');
+          setMajor(value);
+          setMiddle('');
+          setMinor('');
         }}
-        value={selectedMajor}
+        value={majorCategory}
       />
       {/* 중분류 */}
       <FilterDropDown
         defaultValue="중분류"
-        optionValue={middleOptions}
+        optionValue={C.middleOptions[majorCategory] || []}
         onValueChange={(value) => {
-          setSelectedMiddle(value);
-          setSelectedMinor('');
+          setMiddle(value);
+          setMinor('');
         }}
-        value={selectedMiddle}
-        className={cn(defaultFilterStyle, !selectedMajor ? 'pointer-events-none' : '')}
-        mainTextStyle={!selectedMajor ? 'text-gray-400' : ''}
+        value={middleCategory}
+        className={cn(defaultFilterStyle, !majorCategory ? 'pointer-events-none' : '')}
```

Comment:
```suggestion
        className={cn(defaultFilterStyle, majorCategory || 'pointer-events-none')}
```

불필요한 not 연산은 본래 의도를 알기 어렵게 하므로 최대한 간결하게 써주세요.